src/components/overlay/index.js

```diff
@@ -0,0 +1,94 @@
+import _ from 'lodash';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {StyleSheet, Image} from 'react-native';
+import {Colors} from '../../style';
+import {PureBaseComponent} from '../../commons';
+import View from '../view';
+
+
+const gradientImage = require('./assets/GradientOverlay.png');
+const OVERLY_TYPES = {
+  DEFAULT: 'default',
+  TOP: 'top',
+  BOTTOM: 'bottom',
+  SOLID: 'solid'
+}
+/**
+ * @description: Overlay view with types (default, top, bottom, solid)
+ * @extends: Image
+ * @extendsLink: https://facebook.github.io/react-native/docs/image
+ */
+export default class Overlay extends PureBaseComponent {
+  static displayName = 'Overlay';
+
+  static propTypes = {
+    /** 
+    * The type of overlay to set on top of the image 
+    */
+    type: PropTypes.oneOfType([PropTypes.oneOf(Object.keys(OVERLY_TYPES)), PropTypes.string])
+  };
+
+  static overlayTypes = OVERLY_TYPES;
+
+  getStyleByType() {
+    switch (this.props.type) {
+      case OVERLY_TYPES.TOP: 
+        return styles.top;
+      case OVERLY_TYPES.BOTTOM: 
+        return styles.bottom;
+      case OVERLY_TYPES.SOLID: 
+        return styles.solid
+      default:
+        break;
+    }
+  }
+
+  renderImage(typeStyle) {
+    const {type, style} = this.props;
+    const image = type !== OVERLY_TYPES.SOLID ? gradientImage : undefined;
+
+    return (
+      <Image
+        style={[style, styles.container, typeStyle, {width: undefined}]}
+        resizeMode={'stretch'}
+        source={image}
+      />
+    );
+  }
+
+  render() {
+    const {style, type} = this.props;
+    const width = _.get(style, 'width');
+    
+    if (type === OVERLY_TYPES.DEFAULT) {
```

Comment:
I would assume the default behavior of overlay is the top overlay.
this is probably what most people want/use.
WDYT?

Reply:
I set the default for the top+bottom. Please note that there is no real default if you don't pass an overlay type you get no overlay

Reply:
true, but passing DEFAULT will give u both, which is kinda weird. 
I mean the name is weird, not the behavior necessarily . 
maybe worth calling it `BOTH` ? as both top and bottom?

Reply:
I don't like the name BOTH, that's why 'default' seemed better. I don't think it's weird, but if you have a better name I don't mind changing it.